tests/aws/aws_kubernetes_ec2.rs

```diff
@@ -0,0 +1,68 @@
+extern crate test_utilities;
+
+use self::test_utilities::utilities::{context, engine_run_test, generate_cluster_id, generate_id, logger};
+use ::function_name::named;
+use qovery_engine::cloud_provider::kubernetes::Kind as KKind;
+
+use qovery_engine::cloud_provider::aws::kubernetes::VpcQoveryNetworkMode;
+use qovery_engine::cloud_provider::aws::kubernetes::VpcQoveryNetworkMode::{WithNatGateways, WithoutNatGateways};
+use qovery_engine::cloud_provider::aws::regions::AwsRegion;
+use qovery_engine::cloud_provider::Kind;
+use std::str::FromStr;
+use test_utilities::aws::{K3S_KUBERNETES_MAJOR_VERSION, K3S_KUBERNETES_MINOR_VERSION};
+use test_utilities::common::{cluster_test, ClusterDomain, ClusterTestType};
+
+pub const AWS_K3S_VERSION: &str = "v1.20.15+k3s1";
+
+#[cfg(feature = "test-aws-infra")]
```

Comment:
Shouldn't we isolate functional tests for EC2 into their own tests folders with their own features / tests flags?
I guess we might support different K8s versions for AWS and EC2 at some point, having a dedicated folder would avoid mix and dirty plumbering work down the road.

Reply:
It makes sense to me but I don't want to spend too much time on plumbing details until we have something working. If it does not require too much effort - you can structure the way you prefer. Otherwise we can delay it.

Reply:
Done